src/shadowbox/server/manager_service.ts

```diff
@@ -168,6 +173,46 @@ export class ShadowsocksManagerService {
     }
   }
 
+  public async setAccessKeyQuota(req: RequestType, res: ResponseType, next: restify.Next) {
+    try {
+      logging.debug(`setAccessKeyQuota request ${JSON.stringify(req.params)}`);
+      const accessKeyId = req.params.id;
```

Comment:
Is it possible that id is undefined?  It doesn't look like we handle that case

Reply:
It is, in which case the repo will return false and we will serve a 404.